src/cargo/core/compiler/build_context/target_info.rs

```diff
@@ -754,3 +758,79 @@ impl RustcTargetData {
         self.target_config(kind).links_overrides.get(lib_name)
     }
 }
+
+/// Structure used to deal with Rustdoc fingerprinting
+#[derive(Debug, Serialize, Deserialize)]
+pub struct RustDocFingerprint {
+    _rustc_verbose_version: String,
+}
+
+impl RustDocFingerprint {
+    /// Returns the version contained by this `RustDocFingerprint` instance.
+    fn version(&self) -> &str {
+        self._rustc_verbose_version.as_str()
+    }
+
+    /// Given the `doc` dir, returns the path to the rustdoc fingerprint file.
+    fn path_to_fingerprint(doc_dir: &Path) -> PathBuf {
+        doc_dir.to_path_buf().join(".rustdoc_fingerprint.json")
```

Comment:
I added it under `doc` since it's much easier and makes kinda sense. The thing is that if the docs haven't been compiled. It makes no sense to have the fingerprint file at all. 
Once we have the folder, we know we need to look at it and pay attention to the fingerprint.

I'm not sure which effects can have the fact of the fingerprint file living under the `doc` folder. But I can move it to `.fingerprint/` if you consider it better.

Reply:
Yea, I would prefer to keep it out of the `doc` directory.  

I think it can detect whether or not it needs to check/create the fingerprint is to check the `CompileMode` in the `BuildConfig` to see if the current mode is `CompileMode::Doc`.